src/Microsoft.OData.Core/ODataMessageReaderSettings.cs

```diff
@@ -196,6 +196,11 @@ public ODataMessageQuotas MessageQuotas
         /// </summary>
         internal bool ThrowOnUndeclaredPropertyForNonOpenType { get; private set; }
 
+        /// <summary>
+        /// True if null values are omitted, per the omit-values parameter in Preference header of the message.
+        /// </summary>
+        internal bool NullValuesOmitted { get; set; }

```

Comment:
>internal [](start = 8, length = 8)

Should we make that public to allow tests in projects using ODL simple injection? Injecting via PreferenceAppliedHeader() can be a bit complicated (to much stuff to mock)

Reply:
I think the NullValuesOmitted is caused by Preference-Applied header parameter value in the specific request. ODataPreferenceHeader(responseMessage).OmitValus = null should do the trick. 
If it were the case for DependencyInjection, we would need to do it with bool flag for convenience, but it is not the case here, right?